pkg/server/authserver/authentication.go

```diff
@@ -169,6 +160,23 @@ var ldapManager = struct {
 func (s *authenticationServer) UserLogin(
 	ctx context.Context, req *serverpb.UserLoginRequest,
 ) (*serverpb.UserLoginResponse, error) {
+	cookie, err := s.userLogin(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	// Set the cookie header on the outgoing response.
+	if err := grpc.SetHeader(ctx, metadata.Pairs("set-cookie", cookie.String())); err != nil {
+		return nil, srverrors.APIInternalError(ctx, err)
+	}
+
+	return &serverpb.UserLoginResponse{}, nil
+}
+
+// userLogin returns an HTTP cookie upon successful authentication.
+// This is an internal helper used by both the gRPC and HTTP login handlers.
+func (s *authenticationServer) userLogin(
```

Comment:
Just an observation - I see that this helper method uses the rpc types (`serverpb.UserLoginRequest`) and error codes. It makes sense now as it helps share code between the current grpc handler and the http handler. 
However, since login/logout would not be proxied via`apiInternalServer` when drpc is enabled (and post migration), _ideally_, these helper methods should be written to be independent of these rpc types and errors to avoid unnecessary type conversions and encode/decode. Else, we would have to do this as a separate exercise post migration.